src/main/java/codersafterdark/reskillable/base/LevelLockHandler.java

```diff
@@ -42,44 +43,27 @@
     private static Map<Class<?>, List<Class<? extends LockKey>>> lockTypesMap = new HashMap<>();
     private static Map<LockKey, Set<FuzzyLockKey>> fuzzyLockInfo = new HashMap<>();
     private static String[] configLocks;
+    private static Map<String, Class<? extends LockKey>> lockNameMap = new HashMap<>();
+
+    static {
+        registerLockKeyName("itemstack", ItemInfo.class);
+        registerLockKeyName("mod", ModLockKey.class);
+        registerLockKeyName("nbt", GenericNBTLockKey.class);
+    }
 
     public static void loadFromConfig(String[] configValues) {
         configLocks = configValues;
     }
 
     public static void setupLocks() {
         registerDefaultLockKeys();
-        if (configLocks != null) {
```

Comment:
Wasn't this code for pulling from the config file?

Reply:
I looked into how we are doing it currently and the setupLocks method is called in `FMLPostInitializationEvent` so we use it to register the lock keys, and then if the config has locks we register them. We previously loaded the strings here https://github.com/Coders-After-Dark/Reskillable/blob/develop/1.12.2/src/main/java/codersafterdark/reskillable/base/ConfigHandler.java#L53 Looking at the new code the locks are loaded from json here (which makes sense as all custom locks should have been registered by now from addons). I think this is fine however I think the linked line and `LevelLockHandler#loadFromConfig` and `private static  String[] configLocks;` should probably be removed as they are no longer used. That or an automated converter should be made to port existing config files to the JSON format. I will mark through the review the lines that should be able to be deleted.